core/src/tests/dummy_muxer.rs

```diff
@@ -0,0 +1,78 @@
+extern crate futures;
+
+use std::io::Error as IoError;
+use ::muxing::StreamMuxer;
+use futures::prelude::*;
+
+pub struct DummySubstream {}
+
+pub struct OutboundSubstream {}
+
+#[derive(Debug, PartialEq)]
+pub enum DummyConnectionState {
+    Substream,
```

Comment:
What do the `Substream` and `Open` variants mean? Once a substream is open, the substream is yielded as a `DummySubstream` and the state should reset to `Pending`.